api/src/test/java/org/apache/iceberg/transforms/TestBucketingProjection.java

```diff
@@ -0,0 +1,329 @@
+                                          Expression.Operation expectedOp) {
+    Expression projection = Projections.strict(spec).project(filter);
+    Assert.assertEquals(projection.op(), expectedOp);
+  }
+
+  public void assertProjectionInclusiveValue(PartitionSpec spec, UnboundPredicate<?> filter,
+                                             Expression.Operation expectedOp) {
+    Expression projection = Projections.inclusive(spec).project(filter);
+    Assert.assertEquals(projection.op(), expectedOp);
+  }
+
+  public void assertProjectionInclusive(PartitionSpec spec, UnboundPredicate<?> filter,
+                                        Expression.Operation expectedOp, String expectedLiteral) {
+    Expression projection = Projections.inclusive(spec).project(filter);
+    UnboundPredicate<?> predicate = assertAndUnwrapUnbound(projection);
+
+    Assert.assertEquals(predicate.op(), expectedOp);
+
+    Bucket transform = (Bucket) spec.getFieldsBySourceId(1).get(0).transform();
+    if (predicate.op() == Expression.Operation.IN || predicate.op() == Expression.Operation.NOT_IN) {
+      Iterable<?> values = Iterables.transform(predicate.literals(), Literal::value);
+      String actual = Lists.newArrayList(values).stream().sorted()
+          .map(v -> transform.toHumanString(v)).collect(Collectors.toList()).toString();
+      Assert.assertEquals(expectedLiteral, actual);
+    } else {
+      Literal literal = predicate.literal();
+      String output = transform.toHumanString(literal.value());
+      Assert.assertEquals(expectedLiteral, output);
+    }
+  }
+
+  @Test
+  public void testBucketIntegerStrict() {
+    Integer value = 100;
+    Schema schema = new Schema(optional(1, "value", Types.IntegerType.get()));
+    PartitionSpec spec = PartitionSpec.builderFor(schema).bucket("value", 10).build();
+
+    assertProjectionStrict(spec, notEqual("value", value), Expression.Operation.NOT_EQ, "6");
+    assertProjectionStrictValue(spec, equal("value", value), Expression.Operation.FALSE);
+    assertProjectionStrictValue(spec, lessThan("value", value), Expression.Operation.FALSE);
+    assertProjectionStrictValue(spec, lessThanOrEqual("value", value), Expression.Operation.FALSE);
+    assertProjectionStrictValue(spec, greaterThan("value", value), Expression.Operation.FALSE);
+    assertProjectionStrictValue(spec, greaterThanOrEqual("value", value), Expression.Operation.FALSE);
+
+    assertProjectionStrict(spec, notIn("value", value - 1, value, value + 1),
+        Expression.Operation.NOT_IN, "[6, 7, 8]");
+    assertProjectionStrictValue(spec, in("value", value, value + 1), Expression.Operation.FALSE);
+  }
+
+  @Test
+  public void testBucketIntegerInclusive() {
+    Integer value = 100;
+    Schema schema = new Schema(optional(1, "value", Types.IntegerType.get()));
+    PartitionSpec spec = PartitionSpec.builderFor(schema).bucket("value", 10).build();
+
+    assertProjectionInclusive(spec, equal("value", value), Expression.Operation.EQ, "6");
```

Comment:
Am assuming the hash/bucket of the 100 value is 6? Probably add a comment?

Reply:
Thanks for the comments. Added them.